Update GRO525 for A25

MarcOlivierFecteau opened the pull request.

Commit:
Remove ROS package for web interface

Commit:
Update GRO525 for A25

This commit updates the baseline code for GRO525. Because the
`v4l2_camera` package has replaced the custom node for streaming the
camera, there is no reason for `racecar_web_interface/` to be a ROS
package, so it has been migrated to a "vanilla" web application folder.

NOTES:

- This commit does NOT contain any solution: only the (minimum)
  `rosbridge/roslibjs` boilerplate code is present;
- The updated launch file does NOT currently work, because the updated
  serial communication packages have yet to be merged. However, the
  launch file has been tested (succesfully) with the uncomitted changes;
- The `Bootstrap` version used is an older version. I think we SHOULD
  consider migrating to the most recent version of `Bootstrap` for A25.

CHANGES:

- Move "custom script" section into a JavaScript file to follow best
  practices;
- Move dependencies import into `head` to follow best practices;
- Use more explicit names where applicable;
- Add a `clang-format` file for a more uniform code style across the
  repository;
- Update the launch file for GRO525.

chameau5050 commented on racecar_bringup/launch/gro525.launch.py:
```diff
@@ -1,41 +1,68 @@
+)
+
+host_address_arg = DeclareLaunchArgument(
+    "host_address", default_value="127.0.0.1", description="Address of the RaspberryPi"
```
I would keep the 10.42.0.1 for simplicity since it will be easier for student to not worry about the IP address

chameau5050 commented on racecar_bringup/launch/gro525.launch.py:
```diff
@@ -1,41 +1,68 @@
+)
+
+teleop_ld = IncludeLaunchDescription(
+    PythonLaunchDescriptionSource(f"{racecar_bringup_dir}/launch/teleop.launch.py"),
```
it is a better practice to use os.path.join

chameau5050 commented on racecar_web_interface/css/dashboard.css:
```diff
@@ -1,5 +1,8 @@
+html,
```
you updated the lib ?

MarcOlivierFecteau replied:
I didn't update the library's CSS (it is, and SHOULD be, read-only). I only added `html` selector to the `body` selector, and used what I consider to be better defaults. The rest of `dashboard.css` remains unchanged.

chameau5050 commented on .clang-format:
Those change are not related to the Current PR

MarcOlivierFecteau replied:
Indeed, it is _technically_ not related to GRO525. However, `clang-format` was used to format `main.js`.

chameau5050 reviewed (changes requested):
Good job ! Almost there

Commit:
Apply suggested changes for GRO525

CHANGES:

- Use `os.path.join` instead of f-strings for included launch description;
- Revert default IPv4 to `10.42.0.1`.
